doc/api/crypto.md

```diff
@@ -1046,6 +1046,76 @@ encoding of `'utf8'` is enforced. If `data` is a [`Buffer`][], `TypedArray`, or
 
 This can be called many times with new data as it is streamed.
 
+## Class: KeyObject
+<!-- YAML
+added: REPLACEME
+-->
+
+Node.js uses an internal `KeyObject` class which should not be accessed
+directly. Instead, factory functions exist to create instances of this class
+in a secure manner, see [`crypto.createSecretKey`][],
+[`crypto.createPublicKey`][] and [`crypto.createPrivateKey`][]. A `KeyObject`
+can represent a symmetric or asymmetric key, and each kind of key exposes
+different functions.
+
+Most applications should consider using the new `KeyObject` API instead of
+passing keys as strings or `Buffer`s due to improved security features.
+
+### keyObject.getType()
+<!-- YAML
+added: REPLACEME
+-->
+* Returns: {string}
+
+Depending on the type of this `KeyObject`, this function either returns
+`'secret'` for symmetric keys, `'public'` for public (asymmetric) keys or
+`'private'` for private (asymmetric) keys.
+
+### keyObject.getSymmetricSize()
+<!-- YAML
+added: REPLACEME
+-->
+* Returns: {number}
+
+For symmetric keys, this function returns the size of the key in bytes. This
+function is undefined for asymmetric keys.
+
+### keyObject.getAsymmetricKeyType()
+<!-- YAML
+added: REPLACEME
+-->
+* Returns: {string}
+
+For asymmetric keys, this function returns the type of the embedded key (e.g.,
+`'rsa'` or `'dsa'`). This function is undefined for symmetric keys.
```

Comment:
maybe 'dh' or 'ec'? sometimes keys are intended for only dsa or ecdsa, and not key agreement, but I'm not sure that's a meaningful distinction